engine/engine.go

```diff
@@ -428,16 +428,19 @@ func (e *Engine) publishAdvForIndex(ctx context.Context, contextID []byte, metad
 			return cid.Undef, err
 		}
 
-		// Provide the CID link to the content entries to give the indexer the
-		// option to remove all individual entries.  If a contextID is given
-		// then the indexer will delete all content for that contextID and not
-		// need to retrieve the entries.
-		cidsLnk = cidlink.Link{Cid: c}
+		// Create an advertisement to delete content by contextID by specifying
+		// that advertisement has no entries.
+		cidsLnk = schema.NoEntries
```

Comment:
Could you help me understand the rationale behind this change please?
Does this change not assume that ContextID is unique per advertisement? Otherwise, this would wipe all the indices associated to the ContextID; no?

Am I right to assume ContextID [identifies metadata](https://github.com/filecoin-project/go-indexer-core/blob/main/value.go#L15) and not advertisement?

Reply:
When indexed content associated with a context ID, to delete specific indexes then advertise entries that specify the indexes to delete.  When deleting _all_ content associated with a context ID, explicitly do not include entries.

This change does not make any assumption about whether a context ID is unique to an advertisement.

Yes, this change will wipe out all the indices associated with a context.  The comment specifies how to do otherwise.